nvim/msgpack_rpc_stream.lua

```diff
@@ -80,8 +80,51 @@ function MsgpackRpcStream:read_start(request_cb, notification_cb, eof_cb)
     local pos = 1
     local len = #data
     while pos <= len do
-      type, id_or_cb, method_or_error, args_or_result, pos =
-        self._session:receive(data, pos)
+      local helper = function()
+        local a, b, c, d, e = self._session:receive(data, pos)
+        return {a, b, c, d, e}
+      end
+      local status, rv = pcall(helper)
```

Comment:
Try this:

    status, type, id_or_cb, method_or_error, args_or_result, pos = pcall(self._session:receive, data, pos)

`id_or_cb, method_or_error, args_or_result, pos` will all be `nil` if the call fails. Otherwise they'll have the return values.

`type` will be the error message.